src/mini_batch.jl

```diff
@@ -135,3 +143,25 @@ function reassign_labels(DMatrix, metric, labels, centres)
     end
     return labels
 end
+
+"""
+    create_containers(::MiniBatch, k, nrow, ncol, n_threads)
+
+Internal function for the creation of all necessary intermidiate structures.
+
+- `centroids_new` - container which holds new positions of centroids
+- `centroids_cnt` - container which holds number of points for each centroid
+- `labels` - vector which holds labels of corresponding points
+- `sum_of_squares` - vector which holds the sum of squares values for each thread
+"""
+function create_containers(::MiniBatch, X, k, nrow, ncol, n_threads)
+    # Initiate placeholders to avoid allocations
+    T = eltype(X) 
+    centroids_new = Matrix{T}(undef, nrow, k)  # main centroids
+    centroids_cnt = Vector{T}(undef, k)  # centroids counter
```

Comment:
I guess you do not need these two containers (`centroids_cnt` and `centroids_new`)? Try to remove them and see what happens.

Reply:
both removed now. 1 less allocation. Next push will have it deleted.